specs/lib/getUniqueMatchSpec.js

```diff
@@ -22,3 +25,36 @@ describe('getUniqueMatch', function() {
         expect(getUniqueMatch(t2, /[a-z]/g)).toEqual(['a', 'b', 'c']);
     });
 });
+
+describe('Google Group regex', function() {
+    var comments = [];
+    beforeEach(function() {
+        comments = [
+            '',
+            'this has no google link',
+            'https://groups.google.com/',
+            'http://groups.google.com/',
+            'https://groups.google.com/forum/?hl=en#!topic/cesium-dev/x02lygB7hYc',
+            'https://groups.google.com/forum/?hl=en#!topic/cesium-dev/x02lygB7hYc',
+            'Reported here: https://groups.google.com/forum/?hl=en#!topic/cesium-dev/fewafjdsk\n\nSeen with Chrome 59.0.3071.102. iOS version: 10.3.2. Perhaps a driver issue?\n',
+            'https://groups.google.com/forum/?hl=en#!topic/cesium-dev/test1, https://groups.google.com/forum/?hl=en#!topic/cesium-dev/test2'];
+    });
+
+    it('Finds correct number of unique links', function() {
+        expect(getUniqueMatch(comments, googleLinkRegex).length).toEqual(6);
+    });
+
+    it('Returns links intact', function() {
+        expect(getUniqueMatch([comments[2]], googleLinkRegex)).toEqual([comments[2]]);
```

Comment:
Writing tests that index into arrays like this make them hard to debug.  I would make a variable for each item in the `comments` array (each with a descriptive name, for example `surroundedText` and then use the variables instead of `comments[x]`.